mirrord/layer/src/lib.rs

```diff
@@ -141,36 +141,50 @@ pub(crate) fn port_debug_patch(addr: SocketAddr) -> bool {
     }
 }
 
+/// Loads mirrord configuration and applies [`nix_devbox_patch`] patches.
+fn layer_pre_initialization() -> Result<(), LayerError> {
+    let args = std::env::args().collect::<Vec<_>>();
+    let given_process = args
+        .first()
+        .and_then(|arg| arg.split('/').last())
+        .ok_or(LayerError::NoProcessFound)?;
+
+    let mut config = std::env::var("MIRRORD_CONFIG_FILE")
+        .map(PathBuf::from)
+        .map(|path| LayerFileConfig::from_path(&path))?
+        .unwrap_or_default()
+        .generate_config()?;
+
+    nix_devbox_patch(&mut config);
+    let skip_processes = config.skip_processes.clone().map(VecOrSingle::to_vec);
+
+    if should_load(given_process, skip_processes) {
+        layer_start(config);
+    }
+
+    Ok(())
+}
+
+/// The one true start of mirrord-layer.
+#[cfg(not(test))]
 #[ctor]
-fn before_init() {
-    if !cfg!(test) {
-        let args = std::env::args().collect::<Vec<_>>();
-        let given_process = args.first().unwrap().split('/').last().unwrap();
-
-        let config = std::env::var("MIRRORD_CONFIG_FILE")
-            .ok()
-            .and_then(|val| val.parse::<PathBuf>().ok())
-            .map(|path| LayerFileConfig::from_path(&path).unwrap())
-            .unwrap_or_default()
-            .generate_config();
-
-        match config {
-            Ok(mut config) => {
-                nix_devbox_patch(&mut config);
-                let skip_processes = config.skip_processes.clone().map(VecOrSingle::to_vec);
-
-                if should_load(given_process, skip_processes) {
-                    init(config);
-                }
-            }
-            Err(err) => {
-                panic!("Failed to load config: {}", err);
-            }
-        }
+fn mirrord_layer_entry_point() {
+    if let Err(fail) = layer_pre_initialization() {
+        eprintln!(
+            r"
+            mirrord-layer: Encountered unrecoverable error during initialization!
+
+            >> Detailed error: {fail:#?}
+            "
+        );
+        std::process::exit(-1)
```

Comment:
dont think we can use eprint here again, ill have to check with an arm64e binary

Reply:
No more prints!